### PR TITLE
Initial repo baseline

### PR DESCRIPTION
**Pull request by pablocampogo**
Creates initial repo baseline, interfaces and CI.

### CODE REVIEW

File: indexer.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+}
+
+// Indexer struc handler for Indexer functions
+type Indexer struct {
+	provider Provider
+	database Database
+}
+
+// NewIndexer returns Indexer instance with given input
+func NewIndexer(provider Provider, database Database) *Indexer {
```
Comment (rem1niscence) [inline indexer.go:21]: database is locking the users to think it must explicitly be a database, I think a `writer` interface fits better the purpose

File: indexer.go [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+	return &Indexer{
+		provider: provider,
+		database: database,
+	}
+}
+
+// IndexBlock transfers all transactions to a known strcuture and saves them
+func (i *Indexer) IndexBlock(block int) error {
```
Comment (rem1niscence) [inline indexer.go:29]: `IndexBlock` name and a `block` argument makes the naming redundant, could be simply called index and you'll know is a block by the argument

**Review by rem1niscence — CHANGES_REQUESTED** [review]

File: indexer.go (hunk 2 above)
Comment (crisog) [reply indexer.go:29]: The param should be called `block_number`

Comment (rem1niscence) [reply indexer.go:29]: `block_number` is also redundant and using pascal case, we know block is of type number so no need to add it to the name

Comment (crisog) [reply indexer.go:29]: Pascal case slipped thru there, so dismiss that. 

Thing is that I don't believe it is redundant. A block is more complex than a number, even if the types specifies that it should be a number. We should use the appropriate term for the argument, which is a given block _number_.

This is `pocket-go` code, and I think that's totally fine.
```
func (p *Provider) GetBlock(blockNumber int) (*GetBlockOutput, error) {}
```

Comment (Evalir) [reply indexer.go:29]: If y'all disagree about the naming, the _real_ and correct name is `blockHeight`, as that is the name used on `pocket-core`. I do think we should keep the `block` in the name.

Comment (crisog) [reply indexer.go:29]: > If y'all disagree about the naming, the _real_ and correct name is `blockHeight`, as that is the name used on `pocket-core`. I do think we should keep the `block` in the name. Not everyone seeing this code will have so much context on what this is doing.

Agreed

**Review by crisog — APPROVED** [review]

**Review by Evalir — APPROVED** [review]
lgtm! straightforward so far

**Pull request merged**